File: notes/python_vs_go/if/if.go
package main

// if 布尔表达式 {
//    /* 在布尔表达式为 true 时执行 */
// } else {
//   /* 在布尔表达式为 false 时执行 */
// }

// go 没有 elif 这种语法
// if 语句中嵌套 else if...else 语句

func main() {
	for i := 0; i < 10; i++ {
		if i < 5 {
			println(i, "is less than 5")
		} else if i == 5 {
			println(i, "is equal to 5")
		} else {
			println(i, "is greater than 5")
		}
	}

	for i := 0; i < 10; i++ {
		if i > 3 && i < 5 {
			println(i, "is between 3 and 5")
		}
	}
}

File: notes/python_vs_go/if/if.py
# If 语句：
for x in range(10):
    if x < 5:
        print(x, 'is less than 5')
    elif x == 5:
        print(x, 'is equal to 5')
    else:
        print(x, 'is greater than 5')

# python中可以用连续的比较，比如：
for x in range(10):
    if 3<x<5:
        print(x, 'is between 3 and 5')

# python 中还可以使用 if else 语句进行赋值，等同于三元运算符
for x in range(10):
    y= x+10 if x<5 else x+100
    print(y, end=' ')